src/test/java/seedu/address/logic/commands/EnrollCommandTest.java

```diff
@@ -0,0 +1,134 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.participation.Participation;
+import seedu.address.model.person.Person;
+import seedu.address.model.tutorial.Tutorial;
+import seedu.address.testutil.PersonBuilder;
+
+public class EnrollCommandTest {
+
+    private Model testModel;
+    private Person testStudent;
+    private Tutorial testTutorial;
+    private Participation testParticipation;
+
+    @BeforeEach
+    public void setUp() {
+        // Initialize the model with empty lists
+        testModel = new ModelManager();
+
+        // Create test data for a student and a tutorial
+        testStudent = new PersonBuilder().build();
+        testTutorial = new Tutorial("physics");
+
+        // Add the student and tutorial to the model
+        testModel.addPerson(testStudent);
+        testModel.createTutorial(testTutorial);
+
+        // Create the participation record
+        testParticipation = new Participation(testStudent, testTutorial);
+    }
+
+    @Test
+    public void execute_successfulEnrollment() throws CommandException {
+        // EnrollCommand with a valid index and existing tutorial subject
+        EnrollCommand command = new EnrollCommand(Index.fromZeroBased(0), "physics");
+
+        // Check for successful command execution
+        CommandResult result = command.execute(testModel);
+
+        // Check that participation is added successfully
+        assertEquals(String.format(EnrollCommand.MESSAGE_SUCCESS, testStudent.getFullName(), testTutorial.getSubject()),
+                result.getFeedbackToUser());
+        assertTrue(testModel.hasParticipation(testParticipation));
+    }
+
+    @Test
+    public void execute_invalidIndex_throwsCommandException() {
+        // Create an EnrollCommand with an invalid index (out of bounds)
+        EnrollCommand command = new EnrollCommand(Index.fromZeroBased(1), "physics");
```

Comment:
Good to see that the test cases are quite comprehensive and follows the technique covered during lectures.